src/main/scala/com/fulcrumgenomics/rnaseq/CollectErccMetrics.scala

```diff
@@ -0,0 +1,294 @@
+    |[here](https://assets.thermofisher.com/TFS-Assets/LSG/manuals/cms_095046.txt).  The second column lists the ERCC
+    |transcript identifier which should be present as a sequence line in the BAM header, while columns four and five
+    |give the concentration of the transcript for mixtures #1 and #2.
+    |
+    |The choice of mixture to use can be specified with the `--mixture-name` option (either `Mix1` or `Mix2`), or with
+    |a file containing a custom list of transcripts and concentrations using the `--custom-mixture` option as follows.
+    |The custom mixture file should be tab-delimited file containing the following columns:
+    |  1. ERCC ID - each ERCC ID should match a contig/reference-sequence name in the input SAM/BAM header.
+    |  2. Concentration - the concentration (in `attomoles/ul`).
+    |
+    |Three outputs will be produced:
+    |  1. <output>.ercc_summary_metrics.txt - summary statistics for total # of reads mapping to the ERCC transcripts and dose
+    |                                response metrics.
+    |  2. <output>.ercc_detailed_metrics.txt - gives a per-ERCC-transcript expected concentration and observed fragment count.
+    |  3. <output>.ercc_metrics.pdf - plots the expected concentration versus the observed fragment count.
+    |
+    |Secondary, supplementary and fragment reads will be ignored.  A read pair mapping to an ERCC transcript is counted if both
+    |ends of the pair map to the same ERCC transcript.  A minimum mapping quality can be required for reads to be
+    |counted as mapped to an ERCC transcript.
+  """)
+class CollectErccMetrics
+(@arg(flag='i', doc="Input SAM or BAM file of aligned reads in coordinate order.") val input: PathToBam,
+ @arg(flag='o', doc="Output prefix.") val output: PathPrefix,
+ @arg(flag='m', doc="The name of the standard ERCC mixture.", mutex=Array("customMixture")) val mixtureName: Option[ErccMixture] = None,
+ @arg(          doc="Tab-delimited file containing ERCC IDs and expected concentrations.", mutex=Array("mixtureName")) val customMixture: Option[FilePath] = None,
+ @arg(flag='c', doc="Minimum # of counts required to include an ERCC transcript.") val minTranscriptCount: Int = 3,
+ @arg(flag='M', doc="The minimum mapping quality") val minimumMappingQuality: Int = 10
+) extends FgBioTool with LazyLogging {
+
+  Io.assertReadable(input)
+  customMixture.foreach(Io.assertReadable)
+  Io.assertCanWriteFile(output)
+
+  if (mixtureName.isEmpty && customMixture.isEmpty) {
+    throw new ValidationException("Either --mixture-name or --mixture-metadata")
+  }
+
+  private val ScriptPath = "com/fulcrumgenomics/rnaseq/CollectErccMetrics.R"
+
+  override def execute(): Unit = {
+    val in       = SamSource(input)
+    val progress = ProgressLogger(logger, unit=2e6.toInt)
+    val erccData = customMixture match {
+      case Some(path) => DelimitedDataParser(path, '\t').map { row => row[String](0) -> row[Double](1) }.toMap
```

Comment:
Just realized that if the file does _not_ contain headers that this will eat the first data row because `DelimitedDataParser` always expects a header row.